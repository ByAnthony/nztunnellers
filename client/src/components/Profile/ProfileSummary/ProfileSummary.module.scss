@use 'src/index.scss';

.overview {
  margin-top: index.$m-margin;
  font-weight: 500;
  color: index.$text-color;
  text-align: left;

  p {
    margin: 0;
  }

  .fullwidth-main-card {
    @include index.fullwidth-main-card;
  }

  .halfwidth-cards-container {
    @include index.halfwidth-cards-container;
  }

  .halfwidth-secondary-card {
    background-color: index.$primary-medium-color;

    @include index.halfwidth-card-style;
    @include index.halfwidth-secondary-card;
  }

  .image-card {
    line-height: 0;

    img {
      width: 100%;
      border-radius: index.$border-radius;
    }

    @include index.tablet {
      margin: auto;
      width: 50%;
      margin-bottom: index.$xl-margin;
    }
  }
}
