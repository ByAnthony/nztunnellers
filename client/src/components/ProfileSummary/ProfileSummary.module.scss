@use 'src/index.scss';

.header {
  text-align: center;
  margin-bottom: index.$medium-margin;

  @include index.desktop {
    text-align: left;
  }

  @include index.large-desktop {
    text-align: left;
  }
}

.surname {
  display: block;
}

.forename {
  font-weight: 300;
}

.dates {
  margin-top: 50px;
  font-size: 1.25rem;
  font-weight: 300;

  @include index.tablet {
    font-size: 1.5rem;
  }

  @include index.desktop {
    font-size: 1.5rem;
  }

  @include index.large-desktop {
    font-size: 1.625rem;
  }
}

.overview {
  margin-top: index.$large-margin;
  font-weight: 600;
  color: index.$text-color;
  text-align: left;

  @include index.tablet {
    margin-top: index.$extra-large-margin;
  }
}

.fullwidth-main-card {
  @include index.fullwidth-main-card;
}

.halfwidth-cards-container {
  @include index.halfwidth-cards-container;
}

.halfwidth-secondary-card {
  background-color: index.$primary-medium-color;

  @include index.halfwidth-card-style;
  @include index.halfwidth-secondary-card;
}

.halfwidth-secondary-card-title {
  @include index.halfwidth-secondary-card-title;
}

p {
  margin: 0;
}
