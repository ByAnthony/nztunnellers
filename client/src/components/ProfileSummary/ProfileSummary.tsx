import type { EmbarkationUnit, Enlistment, Summary } from '../../types/tunneller';
import STYLES from './ProfileSummary.module.scss';

type props = {
    summary: Summary,
    embarkationUnit: EmbarkationUnit,
    enlistment: Enlistment,
}

export function ProfileSummary({ summary, embarkationUnit, enlistment }: props) {
  const displayBirthDeathDates = (
    birth: string,
    death: string | null,
  ) => (death ? `${birth}-${death}` : `${birth}-â€ ?`);

  const displayUnit = (
    unit: string,
    section: string | null,
  ) => {
    if (unit === 'Main Body' && section) {
      return `${unit}: ${section}`;
    }
    return unit;
  };

  return (
    <>
      <div className={STYLES.header}>
        <h1>
          <span>{ summary.name.forename }</span>
          <span className={STYLES.surname}>{ summary.name.surname }</span>
        </h1>
        <p className={STYLES.dates}>
          { displayBirthDeathDates(summary.birth, summary.death) }
        </p>
      </div>
      <div className={STYLES.overview}>
        <table>
          <tbody>
            <tr>
              <th>Unit</th>
              <th>Rank</th>
              <th>Serial</th>
            </tr>
            <tr>
              <td><p>{ displayUnit(embarkationUnit.detachment, embarkationUnit.section) }</p></td>
              <td><p>{enlistment.rank}</p></td>
              <td><p>{enlistment.serial}</p></td>
            </tr>
          </tbody>
        </table>
      </div>
    </>
  );
}
