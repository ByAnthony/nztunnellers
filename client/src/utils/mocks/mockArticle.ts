import { AboutUs } from '../../types/about-us';
import {
  Article, Image, Next, Section,
} from '../../types/article';

export const mockTitle: string = 'My Awesome\\Article Title';

export const mockChapter: number = 1;

export const mockTopImage: Image = {
  file: 'img-123.png',
  alt: 'Accessible alt text',
};

export const mockArticleImage: Image = {
  ...mockTopImage,
  title: 'An Amazing Photo',
  photographer: 'The French Photographer Doisneau',
  reference: 'Where the photo is preserved',
};

export const mockSectionOne: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockSectionTwo: Section = {
  title: 'Section Title',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea *commodo consequat*. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\nLorem ipsum dolor sit amet, consectetur adipiscing elit,sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat[1](#footnote_1). Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.\\n\\n(6--July 2023)',
};

export const mockNextButton: Next = {
  url: 'my-path-to-next-chapter/',
  title: 'Next\\Chapter',
  chapter: 3,
};

export const mockNotes: string = '[1.](#reference_1) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n\\n[2.](#reference_2) Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

export const mockArticle: Article = {
  id: 'my-awesome-article',
  chapter: mockChapter,
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage, mockArticleImage],
  next: mockNextButton,
  notes: mockNotes,
};

export const mockAboutUs: AboutUs = {
  id: 'my-awesome-article',
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
