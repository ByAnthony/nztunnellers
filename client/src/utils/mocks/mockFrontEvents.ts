import { Date, Event, EventDetail } from '../../types/tunneller';

const mockDate = (year: string | null, dayMonth: string | null): Date => ({
  year,
  dayMonth,
});

export const mockEvent: EventDetail = {
  description: null,
  title: null,
  image: null,
};

export const mockEventCompany: EventDetail = {
  ...mockEvent,
  description: 'Something happened that day for the Company',
  title: 'The Company',
  image: 'image.jpg',
};
export const mockEventDescription: EventDetail = { ...mockEvent, description: 'Something happened that day' };
export const mockEventEnlisted: EventDetail = { ...mockEvent, description: 'Main Body', title: 'Enlisted' };
export const mockEventPosted: EventDetail = { ...mockEvent, description: 'Reinforcement', title: 'Posted' };
export const mockEventTrained: EventDetail = { ...mockEvent, description: 'Military Camp, Somewhere', title: 'Trained' };
export const mockEventTitleAndDescription: EventDetail = { ...mockEvent, description: 'Something happened', title: 'Major Event' };

export const mockEventDiedOfDisease: EventDetail = { ...mockEvent, description: 'Military Hospital', title: 'Died of disease' };
export const mockEventKilledInAction: EventDetail = { ...mockEvent, description: 'Killed on the battlefield', title: 'Killed in action' };
export const mockEventDiedOfWounds: EventDetail = { ...mockEvent, description: 'Wounded on the battlefield', title: 'Died of wounds' };
export const mockEventDiedOfAccident: EventDetail = { ...mockEvent, description: 'Fatal accident', title: 'Died of accident' };
export const mockEventBuried: EventDetail = { ...mockEvent, description: 'Cemetery, France', title: 'Buried' };
export const mockEventGrave: EventDetail = { ...mockEvent, description: '1 A 26', title: 'Grave reference' };

const mockDeathEvent = (deathEvent: EventDetail): Record<string, Event[]> => ({
  1916: [
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [deathEvent, mockEventBuried, mockEventGrave],
    },
  ],
});

export const mockFrontEventsWithDiedOfDisease:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfDisease);

export const mockFrontEventsWithKilledInAction:
    Record<string, Event[]> = mockDeathEvent(mockEventKilledInAction);

export const mockFrontEventsWithDiedOfWounds:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfWounds);

export const mockFrontEventsWithDiedOfAccident:
    Record<string, Event[]> = mockDeathEvent(mockEventDiedOfAccident);

export const mockFrontEvents: Record<string, Event[]> = {
  1915: [
    {
      date: mockDate('1915', '26 January'),
      event: [mockEventEnlisted],
    },
    {
      date: mockDate('1915', '30 January'),
      event: [mockEventPosted],
    },

    {
      date: mockDate('1915', '14 February'),
      event: [mockEventTrained],
    },
    {
      date: mockDate('1915', '4 May'),
      event: [mockEventTitleAndDescription],
    },
  ],
  1916: [
    {
      date: mockDate('1916', '23 June'),
      event: [mockEventDescription, mockEventDescription],
    },
    {
      date: mockDate('1916', '27 September'),
      event: [mockEventCompany],
    },
    {
      date: mockDate('1916', '19 November'),
      event: [mockEventDescription],
    },
    {
      date: mockDate('1916', '12 December'),
      event: [mockEventCompany, mockEventDescription],
    },
  ],
  1917: [
    {
      date: mockDate('1917', '9 April'),
      event: [mockEventTitleAndDescription, mockEventCompany],
    },
    {
      date: mockDate('1917', '8 August'),
      event: [mockEventDescription],
    },
  ],
  1918: [
    {
      date: mockDate('1918', '3 March April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1918', '13 October'),
      event: [mockEventDescription],
    },
  ],
  1919: [
    {
      date: mockDate('1919', '4 April'),
      event: [mockEventTitleAndDescription],
    },
    {
      date: mockDate('1919', '12 July'),
      event: [mockEventDescription],
    },
  ],
};
