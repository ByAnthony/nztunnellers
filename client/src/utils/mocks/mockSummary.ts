import { Summary } from '../../types/tunneller';

export const mockSummary: Summary = {
  serial: '1/2345',
  name: {
    forename: 'John',
    surname: 'Doe',
  },
  birth: '1888',
  death: '1975',
};
