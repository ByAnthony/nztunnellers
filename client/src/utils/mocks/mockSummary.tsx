import { Summary } from '../../types/tunneller';

export const testSummary: Summary = {
  serial: '12345',
  name: {
    forename: 'John',
    surname: 'Doe',
  },
  birth: '1888',
  death: '1975',
};
