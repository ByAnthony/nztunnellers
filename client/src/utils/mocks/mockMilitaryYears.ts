import {
  Date,
  DeathPlace,
  Death,
  DeathCause,
  Cemetery,
  Medal,
  MilitaryYears,
  Enlistment,
  EmbarkationUnit,
  Training,
  Transport,
  EndOfService,
  Demobilization,
} from '../../types/tunneller';
import { mockFrontEvents } from './mockFrontEvents';

export const mockBritishWarMedal: Medal = {
  name: 'British War Medal',
  country: 'United Kingdom',
  image: 'british-war-medal.png',
  citation: null,
};

export const mockVictoryMedal: Medal = {
  name: 'Victory Medal',
  country: 'United Kingdom',
  image: 'victory-medal.png',
  citation: null,
};

export const mockMedals: Medal[] = [
  mockBritishWarMedal,
  mockVictoryMedal,
];

export const mockDate: Date = {
  year: '1926',
  dayMonth: '13 October',
};

const mockPlace: DeathPlace = {
  location: 'Telegraph Hill',
  town: 'Arras',
  country: 'France',
};

const mockCause: DeathCause = {
  type: 'Killed in action',
  circumstances: 'Dead wounded by shell fire',
};

const mockCemetery: Cemetery = {
  name: 'Faubourg d\'Amiens',
  location: 'Arras',
  country: 'France',
  graveReference: 'IE 18',
};

export const mockDeath: Death = {
  date: mockDate,
  place: mockPlace,
  cause: mockCause,
  cemetery: mockCemetery,
  ageAtDeath: 89,
};

export const mockEnlistment: Enlistment = {
  serial: '1/1000',
  rank: 'Sapper',
  date: mockDate,
  district: 'Canterbury',
  alias: null,
  transferredToTunnellers: null,
};

const mockTraining: Training = {
  date: mockDate,
  location: 'Auckland',
  locationType: 'Military Camp',
};

export const mockEmbarkationUnit: EmbarkationUnit = {
  detachment: 'Main Body',
  training: mockTraining,
  section: 'Section No.2',
  attachedCorps: null,
};

export const mockTransport: Transport = {
  reference: 'HMNZT 56',
  vessel: 'Beata',
  departureDate: mockDate,
  departurePort: 'Auckland',
  arrivalDate: mockDate,
  arrivalPort: 'Plymouth',
};

const mockDemobilization: Demobilization = {
  date: mockDate,
  country: 'New Zealand',
};

const mockEndOfService: EndOfService = {
  deserter: false,
  transferred: null,
  deathWar: null,
  transportNz: mockTransport,
  demobilization: mockDemobilization,
};

export const mockMilitaryYears: MilitaryYears = {
  enlistment: mockEnlistment,
  embarkationUnit: mockEmbarkationUnit,
  transportUk: mockTransport,
  frontEvents: mockFrontEvents,
  endOfService: mockEndOfService,
  medals: mockMedals,
};
