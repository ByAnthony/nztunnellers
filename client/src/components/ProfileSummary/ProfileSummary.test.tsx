import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import { mockEmbarkationUnit, mockEnlistment } from '../../utils/mocks/mockMilitaryYears';
import { mockSummary } from '../../utils/mocks/mockSummary';

import { ProfileSummary } from './ProfileSummary';

const component = (
  <ProfileSummary
    summary={mockSummary}
    embarkationUnit={mockEmbarkationUnit}
    enlistment={mockEnlistment}
    image={null}
  />
);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('renders unit and section correctly', () => {
  render(component);

  expect(screen.getByText('Main Body')).toBeInTheDocument();
  expect(screen.getByText('Section No.2')).toBeInTheDocument();
});

test('renders only unit when section unknown', () => {
  const mockComponent = (
    <ProfileSummary
      summary={mockSummary}
      embarkationUnit={{ ...mockEmbarkationUnit, detachment: '1st Reinforcement', section: null }}
      enlistment={mockEnlistment}
      image={null}
    />
  );
  render(mockComponent);

  expect(screen.getByText('1st Reinforcement')).toBeInTheDocument();
  expect(screen.queryByText('Section No.2')).not.toBeInTheDocument();
});

test('renders rank and serial correctly', () => {
  render(component);

  expect(screen.getByText('Sapper')).toBeInTheDocument();
  expect(screen.getByText('1/1000')).toBeInTheDocument();
});

test('renders image when known', () => {
  const mockComponent = (
    <ProfileSummary
      summary={mockSummary}
      embarkationUnit={mockEmbarkationUnit}
      enlistment={mockEnlistment}
      image={{
        url: '1-1000.jpg',
        source: {
          aucklandLibraries: null,
          archives: null,
          family: 'Doe',
          newspaper: null,
          book: null,
        },
      }}
    />
  );
  render(mockComponent);

  expect(screen.getByRole('img').getAttribute('alt')).toEqual('Portrait of John Doe');
  expect(screen.getByRole('img').getAttribute('src')).toEqual('/images/roll/tunnellers/1-1000.jpg');
});
