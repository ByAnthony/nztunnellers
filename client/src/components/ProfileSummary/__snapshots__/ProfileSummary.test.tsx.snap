// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="overview"
  >
    <div
      class="fullwidth-main-card"
    >
      <p>
        Unit
      </p>
      <span>
        Main Body
      </span>
    </div>
    <div
      class="fullwidth-main-card"
    >
      <p>
        Section
      </p>
      <span>
        Section No.2
      </span>
    </div>
    <div
      class="halfwidth-cards-container"
    >
      <div
        class="halfwidth-secondary-card"
      >
        <p>
          Rank
        </p>
        <span>
          Sapper
        </span>
      </div>
      <div
        class="halfwidth-secondary-card"
      >
        <p>
          Serial
        </p>
        <span>
          1/1000
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
